### PR TITLE
Adds slack to pd-cap-recipes so everyone can share!

### PR DESCRIPTION
**Pull request by taavi**
Adding slack notification support!

@robottaway @alperkokmen

### CODE REVIEW

File: lib/pd-cap-recipes/tasks/slack.rb [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      attachment.add_field('Application', slack_application, short: true)
+      attachment.add_field('Environment', slack_environment, short: true)
+      attachment.add_field('Revision', slack_revision(branch), short: true)
+      fetch(:slack_additional_attachments, []).each { |a| attachment.send :add_field, *a }
+      attachment.add_field('Changelog', changelog, short: false) unless changelog.empty?
+
+      webhook = PdSlack::IncomingWebhook.new(
+        fetch(:slack_deploy_incoming_webhook),
+        username: fetch(:slack_username, 'DevOps Taylor Swift'),
+        icon_emoji: fetch(:slack_emoji, ':cooltay:'),
```
Comment (alperkokmen) [inline lib/pd-cap-recipes/tasks/slack.rb]: we should probably document these two in the comments section above as they are going to be fetched as `slack_username` and `slack_emoji`.

for those who want to stick with _tay tay_, here is the image they can use to customize their slack.
image

File: lib/pd-cap-recipes/tasks/slack.rb [hunk 2]
```diff
@@ -0,0 +1,140 @@
+require 'pd-cap-recipes/pd_slack'
+
+Capistrano::Configuration.instance(:must_exist).load do
+  # Slack
+  ########
+  # Please set these in your project's Capfile:
+  # - slack_deploy_incoming_webhook to a posting key authorized for your Slack account, e.g. '/Tfoofoo/Bbarbar/Ibazbazbaz'
+  # - slack_appliation
```
Comment (robottaway) [inline lib/pd-cap-recipes/tasks/slack.rb]: misspelling?

**Comment by robottaway** [comment]
I'm a bit late been under the weather today, but LGTM let's get it shipped and I can update the rt-cap-project to remove Slack functions and then maybe we can update Numerator to test that out?

**Comment by alperkokmen** [comment]
code LGTM – let's give this branch a try with our internal projects and see how it works.

File: lib/pd-cap-recipes/tasks/slack.rb (hunk 2 above)
Comment (taavi) [inline lib/pd-cap-recipes/tasks/slack.rb]: Yep, thanks. :)

**Comment by taavi** [comment]
@alperkokmen I'm not seeing the `Changelog` showing up in the Slack notifications. Walking through the code in my editor doesn't show anything amiss, and I don't see the `rescue` kicking in (and I DO see the `Changelog` header…it just has no bullet points). Thoughts?

**Comment by taavi** [comment]
Fixed the missing changelog. :)

**Comment by robottaway** [comment]
I say merge it. I've tested it end to end with an RT project and it works.

File: lib/pd-cap-recipes/tasks/slack.rb [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  # - slack_deploy_incoming_webhook to a posting key authorized for your Slack account, e.g. '/Tfoofoo/Bbarbar/Ibazbazbaz'
+  # - slack_username is who the slack notification will claim to be from
+  # - slack_emoji is the avatar for who the slack notification will claim to be from
+  # - slack_application
+  # - slack_scm_app_url e.g. 'https://github.com/PagerDuty/appname'
+  # - slack_additional_attachments if you want extra information in the the deploy start message.
+  #   Should be a list of (title, value, short) as per PdSlack::IncomingWebhookAttachment#add_field.
+  # - slack_pub_channel defaults to #deployments-prod (when production) or #deployments-other
+
+  # Hooks
```
Comment (alperkokmen) [inline lib/pd-cap-recipes/tasks/slack.rb]: one last comment – should these be part of configuring slack for your deployment script or set by default? my concern is that some projects may want `before update_code` as opposed to `before deploy` for example.

**Comment by alperkokmen** [comment]
other than [question above](https://github.com/PagerDuty/pd-cap-recipes/pull/56/files#r44456163), LGTM.

File: lib/pd-cap-recipes/tasks/slack.rb (hunk 3 above)
Comment (taavi) [inline lib/pd-cap-recipes/tasks/slack.rb]: I honestly don't know if we should or shouldn't. I value some level of configurability, but I also hate repeating myself in every consuming project.

Comment (alperkokmen) [inline lib/pd-cap-recipes/tasks/slack.rb]: @robottaway @esigler what do you think?

**Pull request merged**